core-bundle/src/Doctrine/Backup/DumperInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Doctrine\Backup;
+
+use Contao\CoreBundle\Doctrine\Backup\Config\CreateConfig;
+use Doctrine\DBAL\Connection;
+
+interface DumperInterface
+{
+    /**
+     * The data yielded is streamed into the backup file. Every yielded string
+     * will be a new line.
+     *
+     * @throws BackupManagerException in case anything went wrong
+     */
+    public function dump(Connection $connection, CreateConfig $config): \Generator;
```

Comment:
Maybe worth annotating that the strings must be returned `@return \Generator<string>`? Is a generator mandatory or do we just need something that is iterable?

Reply:
I wanted to make it mandatory so people make sure, they actually stream stuff. Otherwise one builds a huge array in memory again ;) But I agree with the docs which I have improved in 08d98c8bc21c55a75dc891e465bc77f9be37d37b.

Reply:
Good point with streaming. (Now people are going to write `yield from ArrayIterator($array);` :smile: )

Reply:
Haha, yeah well...it requires enough knowledge about generators to actually **not** do that ;)